### ISSUE TITLE
Bug in 'hipsycl-config.cmake.in'

### ISSUE THREAD

**Issue by breyerml** [AdaptiveCpp/AdaptiveCpp#568]
**Bug summary**
`HIPSYCL_TARGETS` is not set using the environment variable.

**To Reproduce**
`CMakeLists.txt` file:
```
cmake_minimum_required(VERSION 3.19)
project(test LANGUAGES NONE)

find_package(hipSYCL CONFIG REQUIRED)
```
```bash
$ export hipSYCL_DIR=/path/to/cmake/folder
$ export HIPSYCL_TARGETS=cuda:sm_80
$ cmake .
```

**Expected behavior**
`HIPSYCL_TARGETS` should get correctly set to `cuda:sm_80`.

**Optional additional diagnostic information**
```
CMake Error at [...]/lib/cmake/hipSYCL/hipsycl-config.cmake:99 (message):
  More than one hipSYCL target platform is available.

  Must specify desired target(s) [and associated device(s)] via
  HIPSYCL_TARGETS, in the format
  HIPSYCL_TARGETS=<platform1>[:arch1[,arch2][..,archN]][;<platform2>[:arch1][...]][..;<platformN>]
  where platforms are one of omp | cuda.
Call Stack (most recent call first):
  CMakeLists.txt:4 (find_package)
```

**Proposed fix**
There are transposed letters in line 40 of the `hipsycl-config.cmake.in` file:
`set(HIPYSCL_TARGETS_ENV $ENV{HIPSYCL_TARGETS})` should be <br> `set(HIPSYCL_TARGETS_ENV $ENV{HIPSYCL_TARGETS})`

**Comment by fodinabor** [comment]
Oups.. would you mind creating a pull request for this? :)

**Comment by breyerml** [comment]
Done. #569

**Comment by illuhad** [comment]
Fixed by PR #569 - thank you! :)
